File: LeetCode/LeetCode Practice Problems/1464- Maximum Product of Two Elements in an Array.cpp
class Solution {
public:
    int maxProduct(vector<int>& nums) {
        int mx = 0;
        int n = nums.size();
        
        for(int i=0;i<n;i++) {
            for(int j=i+1;j<n;j++) {
                mx = max(mx , (nums[i] - 1) * (nums[j] - 1));
            }
        }
        
        return mx;
    }
};

File: LeetCode/LeetCode Practice Problems/1464-Maximum Product of Two Elements in an Array.java
class Solution {
    public int maxProduct(int[] nums) {
        int mx1 = 0 , mx2 = 0;
        
        for(int num: nums) {
            if(num > mx1) {
                mx2 = mx1;
                mx1 = num;
            }
            else if(num > mx2) {
                mx2 = num;
            }
        }
        
        return (mx1 - 1)*(mx2 - 1);
    }
}
